prov/usnic/src/usdf_cq.c

```diff
@@ -954,9 +716,9 @@ static struct fi_ops_cq usdf_cq_context_ops = {
 static struct fi_ops_cq usdf_cq_context_soft_ops = {
 	.size = sizeof(struct fi_ops_cq),
 	.read = usdf_cq_read_context_soft,
-	.readfrom = usdf_cq_readfrom_context_soft,
+	.readfrom = usdf_cq_readfrom_context,
```

Comment:
My suggestion for `FI_ADDR_NOTAVAIL` is based on this behavior listed in the fi_cq(3) man page:

> The fi_cq_readfrom call allows the CQ to return source address informa-
> tion to the user for any received data.  Source address  data  is  only
> available for those endpoints configured with FI_SOURCE capability.  If
> fi_cq_readfrom is called on an endpoint  for  which  source  addressing
> data   is   not   available,   the   source  address  will  be  set  to
> FI_ADDR_NOTAVAIL.  The number of input src_addr entries  must  the  the
> same as the count parameter.

Returning -FI_ENOSYS is neither an increase nor decrease in functionality, so I don't know how much it matters, but complying w/ the man page seemed like the right thing to do as a tie breaker.